Unable to define XML namespaces directly on controls in XAML

### Current behavior

We are getting an error at the moment when XML namespaces are directly defined on a control

```
1>CSC : error UXAML0001: The type controls:EmbeddedControl could not be found
1>C:\Dev\Samples\MauiSampleApp\MauisampleApp\MainPage.xaml(17,6,17,6): error UXAML0001: An error was found in Grid
1>C:\Dev\Samples\MauiSampleApp\MauisampleApp\MainPage.xaml(11,4,11,4): error UXAML0001: An error was found in StackPanel
1>C:\Dev\Samples\MauiSampleApp\MauisampleApp\MainPage.xaml(1,2,1,2): error UXAML0001: An error was found in Page
1>CSC : error UXAML0001: Processing failed for file C:\Dev\Samples\MauiSampleApp\MauisampleApp\MainPage.xaml (Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Page ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in StackPanel ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Grid ---> System.InvalidOperationException: The type controls:EmbeddedControl could not be found
1>C:\Dev\Samples\MauiSampleApp\MauisampleApp\MainPage.xaml.cs(7,14,7,33): error CS1061: 'MainPage' does not contain a definition for 'InitializeComponent' and no accessible extension method 'InitializeComponent' accepting a first argument of type 'MainPage' could be found (are you missing a using directive or an assembly reference?)
```

Related XAML:
```
<Page x:Class="MauisampleApp.MainPage"
      xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
      xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
      xmlns:local="using:MauisampleApp"
      xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
      xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
      xmlns:maui="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
      xmlns:not_maui="http://notmaui"
      mc:Ignorable="d not_maui"
      Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
  <StackPanel HorizontalAlignment="Center"
              VerticalAlignment="Center">
    <TextBlock AutomationProperties.AutomationId="HelloTextBlock"
               Text="Hello Uno Platform"
               HorizontalAlignment="Center" />
    <maui:Grid>
      <embed:MauiHost x:Name="MauiHostElement"
                      xmlns:embed="using:Uno.Extensions.Maui"
                      xmlns:controls="using:MauisampleApp.MauiControls"
                      Source="controls:EmbeddedControl" />
    </maui:Grid>
    <not_maui:Grid>
      <TextBlock AutomationProperties.AutomationId="NotMauiTextBlock"
                 Text="Alternative content for Non-Maui targets"
                 HorizontalAlignment="Center" />
    </not_maui:Grid>
  </StackPanel>
</Page>
```

### Expected behavior

There should be no error

### How to reproduce it (as minimally and precisely as possible)

[MauiSampleApp.zip](https://github.com/unoplatform/uno/files/14641504/MauiSampleApp.zip)
- Launch the attached sample app
- Try to launch and run the sample app for the maui supported platforms
- Notice the error

### Workaround

Moving the  XML namespaces at the page level for example

### Works on UWP/WinUI

None

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

5.1.50

### Affected platforms

All

### IDE

Visual Studio 2022

### IDE version

Version 17.9.2

### Relevant plugins

None

--- a/src/SourceGenerators/Uno.UI.SourceGenerators.Tests/XamlCodeGeneratorTests/Given_Parser.cs
+++ b/src/SourceGenerators/Uno.UI.SourceGenerators.Tests/XamlCodeGeneratorTests/Given_Parser.cs
@@ -90,7 +90,7 @@ public class Given_Parser
 
 					<Grid>
 
-						<local:MyStackPanel xmlns:SUT="NamespaceUnderTest" Source="SUT:C" />
+						<local:MyStackPanel xmlns:SUT="using:NamespaceUnderTest" Source="SUT:C" />
 
 					</Grid>
 				</Page>

--- a/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs
+++ b/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs
@@ -558,7 +558,7 @@ namespace Uno.UI.SourceGenerators.XamlGenerator
 					{
 						if (ns.Equals(@namespace.Prefix.AsSpan(), StringComparison.Ordinal))
 						{
-							if (_metadataHelper.FindTypeByFullName($"{@namespace.Namespace}.{name.Substring(indexOfColon + 1)}") is INamedTypeSymbol namedType)
+							if (_metadataHelper.FindTypeByFullName($"{GetTrimmedNamespace(@namespace.Namespace)}.{name.Substring(indexOfColon + 1)}") is INamedTypeSymbol namedType)
 							{
 								return namedType;
 							}

